hudi-spark-datasource/hudi-spark3/src/main/scala/org/apache/spark/sql/hudi/catalog/HoodieCatalog.scala

```diff
@@ -192,8 +200,31 @@ class HoodieCatalog extends DelegatingCatalogExtension
     loadTable(ident)
   }
 
+  def deduceTableLocationURI(
```

Comment:
All logic of deduceTableLocationURI is moved from previous HoodieCatalog#createHoodieTable